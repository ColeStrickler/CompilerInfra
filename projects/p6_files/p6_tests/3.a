func : (b: int) -> bool {
	a : int;
	a = 4;
    z : bool;
    x : bool = true;

    return false;
}
