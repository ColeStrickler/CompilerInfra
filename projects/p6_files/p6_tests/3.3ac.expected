[BEGIN GLOBALS]
func
[END GLOBALS]
[BEGIN <init> LOCALS]
[END <init> LOCALS]
fun_<init>: enter <init>
lbl_0:      leave <init>
[BEGIN func LOCALS]
b (formal arg of 8 bytes)
a (local var of 8 bytes)
z (local var of 8 bytes)
x (local var of 8 bytes)
[END func LOCALS]
fun_func:   enter func
            getarg 1 [b]
            [a] := 4
            [x] := 1
            setret 0
            goto lbl_1
lbl_1:      leave func
